### PR TITLE
:bug:  Fix edit migration wave

### PR DESCRIPTION
**Pull request by ibolton336**
Closes https://github.com/konveyor/tackle2-ui/issues/872
- Disable edit date of a previously started migration wave 
- Use dayjs instead of custom formatter/parser 
- Add validation for end date before start date

### CODE REVIEW

**Review by gildub — CHANGES_REQUESTED** [review]
There still are issues. The date selected is not recorded to the same date, I'm not sure if it's because of my TZ.

**Comment by ibolton336** [comment]
> There still are issues. The date selected is not recorded to the same date, I'm not sure if it's because of my TZ.

Ah I think I see the issue. The validation for start date is checking against the current date for existing waves.

**Comment by gildub** [comment]
The dates between table values and edit values don't match: 

image

image

**Review by gildub — CHANGES_REQUESTED** [review]
Same issue

**Review by gildub — COMMENTED** [review]
Looks good, just one minor comment.

**Review by mturley — COMMENTED** [review]
LGTM and works well. One note, which is maybe outside the scope of this fix but related to validation on these fields and is a 2-line fix.

File: client/src/app/pages/migration-waves/components/migration-wave-form.tsx [hunk 1]
```diff
@@ -249,24 +250,30 @@ export const WaveForm: React.FC<WaveFormProps> = ({
...
-                    dateFormat={dateFormat}
-                    dateParse={dateParse}
-                    appendTo={() =>
-                      document.getElementById(
-                        "create-edit-migration-wave-modal"
-                      ) as HTMLElement
-                    }
-                  />
-                )}
+                renderInput={({ field: { value, name, onChange } }) => {
```
[the remark this answers is not in the document]
Comment (ibolton336) [reply client/src/app/pages/migration-waves/components/migration-wave-form.tsx:253]: +1

**Comment by ibolton336** [comment]
Fixed, thanks @mturley

**Review by mturley — APPROVED** [review]
LGTM

**Pull request merged**